src/core/thread/src_match_controller.cpp

```diff
@@ -174,25 +178,25 @@ void SourceMatchController::ClearEntry(Child &aChild)
 
     if (aChild.mSourceMatchPending)
     {
-        otLogDebgMac(GetInstance(), "SrcAddrMatch - Clearing pending flag for 0x%04x", aChild.mValid.mRloc16);
+        otLogDebgMac(GetInstance(), "SrcAddrMatch - Clearing pending flag for 0x%04x", aChild.GetRloc16());
         aChild.mSourceMatchPending = false;
         ExitNow();
     }
 
     if (aChild.mUseShortAddress)
     {
-        error = otPlatRadioClearSrcMatchShortEntry(GetInstance(), aChild.mValid.mRloc16);
+        error = otPlatRadioClearSrcMatchShortEntry(GetInstance(), aChild.GetRloc16());
 
         otLogDebgMac(GetInstance(), "SrcAddrMatch - Clearing short address: 0x%04x -- %s (%d)",
-                     aChild.mValid.mRloc16, otThreadErrorToString(error), error);
+                     aChild.GetRloc16(), otThreadErrorToString(error), error);
     }
     else
     {
-        uint8_t addr[sizeof(aChild.mMacAddr)];
+        uint8_t addr[sizeof(aChild.GetExtAddress())];
```

Comment:
I think it maybe safer to not use the `sizeof(aChild.GetExtAddress())` here and use somethig like `sizeof(Mac::ExtAddress)` instead.

We are getting `sizeof` of return value of a function that returns a reference to `Mac::ExtAddress` (`const Mac::ExtAddress &GetExtAddress()`)... I am not sure if `sizeof` gives the size of the `Mac::ExtAddress` or sizeof a reference to it (which will then be incorrect).